extension/src/popup/views/AccountHistory/index.tsx

```diff
@@ -141,86 +143,72 @@ export const AccountHistory = () => {
       return segments;
     };
 
-    const fetchAccountHistory = async () => {
-      try {
-        const operations = await getAccountHistory(publicKey, networkDetails);
-        setHistorySegments(createSegments(operations));
-      } catch (e) {
-        console.error(e);
-      }
-    };
-
-    const getData = async () => {
-      setIsLoading(true);
-      await fetchAccountHistory();
-      setIsLoading(false);
-    };
-
-    getData();
-  }, [publicKey, networkDetails, dispatch, isHideDustEnabled]);
-
-  return isDetailViewShowing ? (
-    <TransactionDetail {...detailViewProps} />
+    if (getHistoryState.state === RequestState.SUCCESS) {
+      setHistorySegments(createSegments(getHistoryState.data));
+    }
+  }, [
+    getHistoryState.state,
+    getHistoryState.data,
+    publicKey,
+    networkDetails,
+    isHideDustEnabled,
+  ]);
+
+  const showLoader =
+    getHistoryState.state === RequestState.IDLE ||
+    getHistoryState.state === RequestState.LOADING;
+  const hasEmptyHistory = !showLoader || !getHistoryState.data;
```

Comment:
should we be checking for `data.length` here?
do we really need to check the `showLoader` value here?

e.g.:
`const hasEmptyHistory = !getHistoryState?.data?.length;`

Reply:
we can't rely on the history length alone because then we would show this message during loading or idle states. I believe the correct logic is "show this when data is not loading and there is no history".

Reply:
got it, so maybe something like this? 
`const hasEmptyHistory = !showLoader && !getHistoryState?.data?.length;`

Reply:
actually since you suggested we pull out the loading render branch, it did make it easier to use your original suggestions. Updated in c1612f50a5b634df9cc37b607ff19754169872cd